src/foam/nanos/crunch/extra/ImplyReviewedAction.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.extra',
+  name: 'ImplyReviewedAction',
+  implements: ['foam.nanos.ruler.RuleAction'],
+
+  javaImports: [
+    'foam.core.ContextAgent',
+    'foam.core.X',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CrunchService',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'java.util.List',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+      agency.submit(x, new ContextAgent() {
+        @Override
+        public void execute(X x) {
+          // In case this rule action is misconfigured
+          if ( ! (obj instanceof UserCapabilityJunction) ) {
+            throw new RuntimeException(
+              "ImplyReviewedAction can only be triggered on UCJ objects");
+          }
+
+          // Context requirements
+          var capabilityDAO = (DAO) x.get("capabilityDAO");
+
+          var ucj = (UserCapabilityJunction) obj;
+          var cap = (Capability) capabilityDAO.find(ucj.getTargetId());
+
+          // Create UCJ subject context
+          x = x.put("subject", ucj.getSubject(x));
+
+          // Find the review wizard corresponding to this capability.
+          //   A list is used in case there is more than one.
+          List reviewWizards = ((ArraySink) capabilityDAO.where(AND(
```

Comment:
naming suggestion:
ReviewWizard -> ReviewOtherCapCapability

ReviewWizard makes me think client ui.

Reply:
ReviewWizard is the top-level capability for a group of ReviewCapability, so in a way it does represent the wizard.